### ISSUE TITLE
Update/Add Platform Toolset v143 for windows-latest

### ISSUE THREAD

**Issue by aardappel** [actions/virtual-environments#4468]
### Tool name

Platform Toolset v143

### Tool license

no idea

### Add or update?

- [X] Add
- [ ] Update

### Desired version

v143

### Brief description of tool

VS2022 was just released, but it is not supported on windows-latest. I can't use windows-2022 either, since that is missing other tools. Platform Toolset v143 would be sufficient for building thru MSBuild.

### URL for tool's homepage

https://visualstudio.microsoft.com/vs/

### Virtual environments affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] Windows Server 2016
- [X] Windows Server 2019
- [X] Windows Server 2022

### Can this tool be installed during the build?

```bash
That's not what has been done in previous images.
```

### Are you willing to submit a PR?

No

**Comment by dibir-magomedsaygitov** [comment]
Hello @aardappel. Thank you for your request. We've prepared [the PR](https://github.com/actions/virtual-environments/pull/4491) to add `Microsoft.VisualStudio.ComponentGroup.UWP.VC` to VS2022 (v143). Unfortunately, VS2019 doesn't support v143.

**Comment by miketimofeev** [comment]
@aardappel what tools prevent you from using windows-2022?

**Comment by KVROHITH098** [comment]
Well he says that the file which is a system admin is locking him out of the PC database so that is why he is unable to login

**Comment by aardappel** [comment]
Not exactly sure why, but I got this with the 2022 image:

```
Run microsoft/setup-msbuild@v1.0.2
C:\ProgramData\Chocolatey\bin\vswhere.exe -products * -requires Microsoft.Component.MSBuild -property installationPath -latest
Error: Unable to find MSBuild.
```
see run: https://github.com/aardappel/lobster/runs/4170461751?check_suite_focus=true

**Comment by dibir-magomedsaygitov** [comment]
Hello @aardappel. I've tried to reproduce the issue, but step `- name: Add msbuild to PATH` was successful. 
![test](https://user-images.githubusercontent.com/61747324/141737216-2673b7d8-90e9-4ec7-9a82-e1d3d0c0b078.jpg)

**Comment by miketimofeev** [comment]
@aardappel you error was related to the preview version of the Visual Studio 2022, it is fixed in 1.0.3 version of the task https://github.com/microsoft/setup-msbuild/releases/tag/v1.0.3
But since the stable VS version was released you can continue using 1.0.2 without any issues — it should work now as @dibir-magomedsaygitov just checked

**Comment by aardappel** [comment]
Thanks, with `1.0.3` this now builds with `windows-2022`!
